### PR TITLE
return evt result of of flash loading status to callback function

### LINKED ISSUE #4: callback function should invoked with evt object of status of flash loading
add evt parameter to call back function to be same as [api ](https://code.google.com/p/swfobject/wiki/api) ,

> callbackFn is a JavaScript function that has an event object as a parameter:
> success, Boolean to indicate whether the creation of a Flash plugin-in <object> DOM was successful or not
> id, String indicating the ID used in swfobject.registerObject
> ref, HTML object element reference (returns undefined when success=false)

### PR DESCRIPTION
**Pull request by waleedsamy**
fix #4

### CODE REVIEW

**Comment by jeef3** [comment]
Awesome, thanks!

File: angular-swfobject.js [hunk 1]
```diff
@@ -60,23 +60,21 @@ angular.module('swfobject', [])
                   $interval.cancel(loadCheckInterval);
                   loadCheckInterval = null;
                   //Execute function
-                  fn();
+                  fn({evt: evt});
```
Comment (jeef3) [inline angular-swfobject.js:63]: I think this is meant to be just `fn(evt);`?

File: angular-swfobject.js [hunk 2]
```diff
@@ -60,23 +60,21 @@ angular.module('swfobject', [])
...
-          if (scope.swfLoad) {
-            swfLoadEvent(evt, function () {
-              scope.swfLoad();
-            });
+          if (scope.swfLoad && typeof(scope.swfLoad) === "function") { // proceed if there are callback function
+            if (!evt.success || !evt.ref) { // if failure no reason to go and check if flash is 100% loaded 
+              scope.swfLoad({evt: evt});
+            }else{
+              swfLoadEvent(evt, scope.swfLoad);
+            }
```
Comment (jeef3) [inline angular-swfobject.js:78]: - Same here: `scope.swfLoad(evt);`
- Watch formatting/spacing: `} else {`
- Probably don't need the "proceed if there are callback function" comment.
- Keep the "if failure no reason to go and check if flash is 100% loaded" comment, but place it either above the `if` or above the `scope.swfLoad` call.

File: angular-swfobject.js [hunk 3]
```diff
@@ -60,23 +60,21 @@ angular.module('swfobject', [])
...
-                  fn();
+                  fn({evt: evt});
                 }
               }, 1500);
             }
           }, 200);
         }
 
-        //This function is invoked by SWFObject once the <object> has been created
+        //a callback function that is called on both success or failure of creating a Flash plug-in <object> on the page (SWFObject 2.2+)
```
Comment (jeef3) [inline angular-swfobject.js]: You could probably remove this comment altogether and just link to the docs: https://code.google.com/p/swfobject/wiki/api

File: README.md [hunk 4]
```diff
@@ -39,11 +39,17 @@ Specify flash vars:
 ```
 
 Register a callback for swf load complete:
+  update :
+    callback can accept evt option as https://code.google.com/p/swfobject/wiki/api swfobject.embedSWF documentation
+                    callbackFn is a JavaScript function that has an event object as a parameter:
+                      success, Boolean to indicate whether the creation of a Flash plugin-in <object> DOM was successful or not
+                      id, String indicating the ID used in swfobject.registerObject
+                    ref, HTML object element reference (returns undefined when success=false)
```
Comment (jeef3) [inline README.md]: Again, probably don't need to quote the documentation, just link through to it. If you are going to quote it, then the Markdown formatting will need some tidying up.

**Comment by jeef3** [comment]
Thanks again @waleedsamy! Just a couple of minor comments before we can merge this one.

File: angular-swfobject.js (hunk 1 above)
Comment (waleedsamy) [inline angular-swfobject.js:63]: unfortunately this style not work and will be passed as undefined to callback function [as](http://stackoverflow.com/questions/17556703/angularjs-directive-call-function-specified-in-attribute-and-pass-an-argument-to)

Comment (jeef3) [inline angular-swfobject.js:63]: Ah yup, I see what it's doing now. :+1:

File: angular-swfobject.js [hunk 5]
```diff
@@ -60,23 +60,23 @@ angular.module('swfobject', [])
...
-              scope.swfLoad();
-            });
+          if (scope.swfLoad && typeof(scope.swfLoad) === "function") {
+            // if failure no reason to go and check if flash is 100% loaded 
+            if (!evt.success || !evt.ref) {
+              scope.swfLoad({evt: evt});
+            }
+            else{
+              swfLoadEvent(evt, scope.swfLoad);
+            }
```
Comment (jeef3) [inline angular-swfobject.js:78]: @waleedsamy thanks for the fixes. I'm not sure what happened to this one though? What I meant was:

``` JavaScript
if (thing) {
  // code
} else {
  // code
}
```

_Cuddle braces, with spaces between braces and `else`._

**Pull request merged**